driver/registers.c

```diff
@@ -167,15 +174,17 @@ static ssize_t amc_pci_reg_read(
 
     /* In non blocking mode if we're not ready then say so. */
     bool no_wait = file->f_flags & O_NONBLOCK;
-    if (no_wait  &&  !interrupt_events_ready(context->interrupts))
+    if (no_wait  &&  !interrupt_events_ready(context->interrupts,
+                                             context->reader_number))
```

Comment:
Don't like indenting to hanging brackets.  I know that continuation indenting in an `if` is tricky, but don't do like this, please.